### PR TITLE
[redaction] redaction engine throws exception when the query is having non-ascii character

### PR DESCRIPTION
**Pull request by Akhilsnaik**
## What changes were proposed in this pull request?
- #1972 - redaction engine throws exception when the query is having non-ascii  character

## How was this patch tested?

- Tested manually in Python-3 and python-2 server , fix works well
Screenshot 2021-04-04 at 7 52 33 PM

### CODE REVIEW

**Review by romainr — COMMENTED** [review]
Nice!

Is there a way to add a unit test?

**Comment by Akhilsnaik** [comment]
```
asnaik@asnaik-MBP15-2:~/Documents/Code/forked_cloudera_hue/hue|GH-1972⚡ ⇒  build/env/bin/hue test specific desktop.redaction
.
.
.

........................[04/Apr/2021 21:28:22 -0700] tests        DEBUG    message
[04/Apr/2021 21:28:22 -0700] tests        DEBUG    message an arg
[04/Apr/2021 21:28:22 -0700] tests        DEBUG    password="???"
[04/Apr/2021 21:28:22 -0700] tests        DEBUG    password="???"
[04/Apr/2021 21:28:22 -0700] tests        DEBUG    password="???"
[04/Apr/2021 21:28:22 -0700] tests        DEBUG    before password="???" after
[04/Apr/2021 21:28:22 -0700] tests        DEBUG    ssn=XXX-XX-XXXX
.[04/Apr/2021 21:28:22 -0700] tests        DEBUG    Message to redact : äöüß 123-45-6789
[04/Apr/2021 21:28:22 -0700] tests        DEBUG    Message after redact : äöüß XXX-XX-XXXX
[04/Apr/2021 21:28:22 -0700] tests        DEBUG    Message to redact : 你好阿基尔 1234234534654576
[04/Apr/2021 21:28:22 -0700] tests        DEBUG    Message after redact : 你好阿基尔 XXXXXXXXXXXXXXXX
[04/Apr/2021 21:28:22 -0700] tests        DEBUG    Message to redact : ã 你好 1234,2345,3456,4576
[04/Apr/2021 21:28:22 -0700] tests        DEBUG    Message after redact : ã 你好 XXXX-XXXX-XXXX-XXXX
......
----------------------------------------------------------------------
Ran 31 tests in 1.213s

OK
Destroying test database for alias 'default'...
[04/Apr/2021 21:28:22 -0700] test         INFO     Tests (desktop.redaction) returned 0
```

File: desktop/core/src/desktop/redaction/tests.py [hunk 1]
```diff
@@ -332,6 +335,22 @@ def test_real_rules(self):
     for message, redacted_message in messages:
       assert_equal(redacted_message, policy.redact(message))
 
+  def test_unicode_strings(self):
+    path = get_path('real-1.json')
```
Comment (romainr) [inline desktop/core/src/desktop/redaction/tests.py:339]: I see, it is a test file already there

File: desktop/core/src/desktop/redaction/engine.py [hunk 2]
```diff
@@ -106,7 +107,7 @@ def redact(self, message):
     """
 
     if message and (self.trigger is None or self.trigger.search(message)):
-      return self.regex.sub(self.replace, message)
+      return self.regex.sub(smart_str(self.replace), message)
```
Comment (romainr) [inline desktop/core/src/desktop/redaction/engine.py:110]: If you revert the change, does the test fail? :)

**Review by romainr — APPROVED** [review]
Nice!

File: desktop/core/src/desktop/redaction/tests.py (hunk 1 above)
Comment (Akhilsnaik) [reply desktop/core/src/desktop/redaction/tests.py:339]: yes real-1.json are the redaction rules defined json
https://github.com/cloudera/hue/blob/master/desktop/core/src/desktop/redaction/test_data/real-1.json

reusing the same file as it suffice the basic neccessity :)

File: desktop/core/src/desktop/redaction/engine.py (hunk 2 above)
Comment (Akhilsnaik) [reply desktop/core/src/desktop/redaction/engine.py:110]: it has to fail in python-2 as `message` will be processed as `message_to_redact = smart_str(message)` and its a multibyte string.
happen to see the build failing on the first place without abouve change, so i have added those too.
Hopefully no more issue with non-ascii characters from now on in editor.

Comment (romainr) [reply desktop/core/src/desktop/redaction/engine.py:110]: "it has to", so not tested I guess ;)
